.github/workflows/main.yml

```diff
@@ -61,17 +61,17 @@ jobs:
     # so, seperating steps with if conditions
     - name: build ROS2 for foxy
       if: ${{ matrix.ros_distro == 'foxy' }}
-      uses: ros-tooling/setup-ros@v0.6
+      uses: ros-tooling/setup-ros@236ab287884fd5a314fc030e91b2017abb46719e #v0.6
```

Comment:
Where did you get this hash from?
I see here a different one
https://github.com/ros-tooling/setup-ros/releases/tag/v0.6

Reply:
I see where from,
You use 0.63,
Let's try 0.6 as 0.63 failed

Reply:
At first I put 0395475ab6416edc274bb3c43673b2091a217e02 (V0.6) and that caused problems with the FOXY distro.
So now changed it to 236ab287884fd5a314fc030e91b2017abb46719e (0.6.3 last commit on v0.6)